python/dtype.cpp

```diff
@@ -25,7 +24,8 @@ scipp::core::DType scipp_dtype(const py::dtype &type) {
   if (type.is(py::dtype::of<std::int64_t>()) ||
       (type.kind() == 'i' && type.itemsize() == 8))
     return scipp::core::dtype<int64_t>;
-  if (type.is(py::dtype::of<std::int32_t>()))
+  if (type.is(py::dtype::of<std::int32_t>()) ||
+      (type.kind() == 'i' && type.itemsize() == 4))
```

Comment:
Why is this needed now?

Reply:
All linked to the above. i.e on windows doing something like:
```python
var = Variable([Dim.X], np.arange(10))
```
exposes this problem.